### ISSUE TITLE
Outlook for Windows crashes after closing the dialog box

### ISSUE THREAD

**Issue by clavx** [OfficeDev/office-js#1545]
> Message from office-js bot:
We’re closing this issue because it has been inactive for a long time. We’re doing this to keep the issues list manageable and useful for everyone. If this issue is still relevant for you, please create a new issue.
> Thank you for your understanding and continued feedback.

In Outlook for Windows, closing a dialog box using the `close()` function causes Outlook to terminate unexpectedly.

## Expected Behavior

Outlook does not crash.

## Current Behavior

Outlook crashes when we use the On-Send feature.

## Steps to Reproduce, or Live Example

I modified some of the code generated by generator-office as follows to automatically close the dialog screen 5 seconds after it is displayed.
``` ts
let _dialog: Office.Dialog;
function action(event: Office.AddinCommands.Event) {
  Office.context.ui.displayDialogAsync(location.origin + "/dist/taskpane.html", {
    displayInIframe: true,
    width: 40,
    height: 20,
  }, (r) => {
    _dialog = r.value;
  });

  setTimeout(() => {
    _dialog.close();
    event.completed({ allowEvent: false });
  }, 5000);
}
```

## Context

We have seen this happen with the Beta Channel, Current Channel Version 2010 (Build 13328.20356), and Monthly Enterprise Channel Version 2009 (Build 13231.20514).
It did not occur in the Semi-Annual Enterprise Channel Version 2002 (Build 12527.21330).
Also, switching to Edge WebView2 Runtime in Beta Channel stopped the issue.
Note: This is the version we have actually checked.

As a workaround, by exiting without calling the `_dialog.close();` function, the Outlook crash no longer occurs, so that is what I am doing now.
Please let me know if there are any problems without using the `close ()` function.

## Your Environment

* Platform [PC desktop, Mac, iOS, Office on the web]: PC desktop
* Host [Excel, Word, PowerPoint, etc.]: Outlook
* Office version number: Version 2011 (Build 13426.20332)
* Operating System: Windows 10 Version 2004
* Browser (if using Office on the web): N/A

## Useful logs

This is a reproduced video.
It always occurs after about 5 times of sending operation.
This is more likely to occur with the 32-bit version of Outlook than the 64-bit version of Outlook.

![OutlookBug](https://user-images.githubusercontent.com/58804032/101762509-df859d00-3b20-11eb-8687-a09d27232fb1.gif)

**Comment by exextoc** [comment]
I was unable to repro the issue you are seeing. Could you provide us with a crash dump or Watson ID for the crash?

**Comment by clavx** [comment]
This has been observed in the following test environments:
- Windows 10 running on Hyper-V in a local PC
- Windows 10 running on Azure

The Azure VM was created with the following configuration:
- Size: Standard D2s v3 (2 vcpus, 8 GiB memory)
- Operating system: Windows
- Publisher: MicrosoftWindowsDesktop
- Offer: Windows-10
- Plan: 20h1-pro-g2
- VM generation: V2

-----

View technical details
> Source
> Microsoft Outlook
> 
> Summary
> Stopped working
> 
> Date
> ‎12/‎18/‎2020 8:10 AM
> 
> Status
> Report sent
> 
> Description
> Faulting Application Path:	C:\Program Files (x86)\Microsoft Office\root\Office16\OUTLOOK.EXE
> 
> Problem signature
> Problem Event Name:	APPCRASH
> Application Name:	OUTLOOK.EXE
> Application Version:	16.0.13426.20332
> Application Timestamp:	5fcdca06
> Fault Module Name:	OSF.DLL
> Fault Module Version:	16.0.13426.20120
> Fault Module Timestamp:	5fa134a3
> Exception Code:	c0000005
> Exception Offset:	000f6bc4
> OS Version:	10.0.19041.2.0.0.256.48
> Locale ID:	1033
> 
> Extra information about the problem
> Bucket ID:	9c301f7e445663ff98ede8066c22740c (1796346940672341004)
> 

I tried to get a dump with WER referring to the following information, but the file was not created.
I will try to get it with ProcDump

https://support.microsoft.com/en-us/help/4520151/preparing-for-support-calls-for-browser-crashes

**Comment by clavx** [comment]
I was able to get the dump by running ProcDump as follows, but the file size is about 660MB.
How can I share it?

```
procdump.exe -ma -e <PID>
```

**Comment by lindalu-MSFT** [comment]
Hi clavx, can you zip it and place it on your public drive or even try attaching it here? If compressed, you might be able to drag and drop the dump directly to a message. That way, the Outlook eng team can have a closer look.

**Comment by clavx** [comment]
Just in case, I got another account and recreated the Azure environment and tried it, but it reproduced the issue.

https://cube3w-my.sharepoint.com/:u:/g/personal/test_cube3w_onmicrosoft_com/EZqz3kAL7_tChOkIfA7v2wMBjtIhy6rDbXWzP3eZScbh-g?e=5u2BF8

**Comment by lliu113** [comment]
@clavx - thanks for providing the dump! We will look into this internally

**Comment by sagarpomal** [comment]
Hi @clavx, @lliu113,

We are facing a similar issue with Word on Azure VM. The Office Store security team uses VM to test the add-ins, and it is failing because of the issue. Have you found any resolution for it?

**Comment by clavx** [comment]
I checked the current status because the WebView2 was automatically deployed.

It no longer occurs in the Monthly Enterprise Channel.
It is still occurring in the Semi-Annual Enterprise Channel.

OK: Version 2102 (Build 13801.20506) with WebView (EdgeHTML) or WebView2 (Chromium)
NG: Version 2008 (Build 13127.21506)

**Comment by clavx** [comment]
I checked the current status.
I think this issue has been resolved in version 2102.

Semi-Annual Enterprise Channel:
- Resolved: Version 2102 (Build 13801.20808)
- Not resolved: Version 2008 (Build 13127.21792)

Any updates on this?
